### DISCUSSION TITLE
Is there any way to change the HostKeyAlgorithms from Teleport machines to use a library such as ssh2 in NodeJS to open a connection?

### DISCUSSION THREAD

**Discussion by ghost** [gravitational/teleport#42024] [Q&A] [answered]
Hello!

I have a problem and I wanted to know if the community could help me solve it. I need to open an ssh connection to the teleport machines from a Node.JS library, but I'm not able to do so, as I'm receiving the following error: Handshake failed: no matching host key format.

My teleport's ssh configuration is as follows:
```
Host *.teleport teleport.xxx.com
    UserKnownHostsFile "/Users/lucasfellipe/.tsh/known_hosts"
    IdentityFile "/Users/lucasfellipe/.tsh/keys/teleport.xxx.com/lucas.fellipe"
    CertificateFile "/Users/lucasfellipe/.tsh/keys/teleport.xxx.com/lucas.fellipe-ssh/teleport-cert.pub"
    HostKeyAlgorithms rsa-sha2-512-cert-v01@openssh.com,rsa-sha2-256-cert-v01@openssh.com,ssh-rsa-cert-v01@openssh.com

# Flags for all teleport hosts except the proxy
Host *.teleport !teleport.xxx.com
    Port 3022
    ProxyCommand "/usr/local/bin/tsh" proxy ssh --cluster=teleport --proxy=teleport.xxx.com:443 %r@%h:%p
```

And I wanted to know if there is a possibility of changing the key algorithm so I can use the library, since it does not support these algorithms. Can someone help me?

**Comment by webvictim** [comment, marked as answer]
No, Teleport only supports the `HostKeyAlgorithms` listed.
